dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java

```diff
@@ -579,6 +645,38 @@ private void validateNoConflictsWithScheduledExperiments(final Experiment experi
                 DotStateException.class);
     }
 
+    private void cancelScheduledExperimentsUponConflicts(final Experiment experimentToCheck,
+            final User user) throws DotDataException {
+
+        final List<Experiment> scheduledExperimentsOnPage = list(ExperimentFilter.builder()
+                .pageId(experimentToCheck.pageId())
+                .statuses(Set.of(SCHEDULED))
+                .build(), user);
+
+        scheduledExperimentsOnPage.stream().filter(scheduledExperiment -> {
+            final Scheduling scheduling = scheduledExperiment.scheduling().orElseThrow();
+            final Scheduling schedulingToCheck = experimentToCheck.scheduling().orElseThrow();
+            return conflictingStartOrEndDate(scheduling, schedulingToCheck);
+        }).forEach(scheduledExperiment -> {
+            try {
+                cancel(scheduledExperiment.id().orElseThrow(), user);
+            } catch (DotDataException | DotSecurityException e) {
+                throw new DotStateException(e);
+            }
+        });
+    }
+
+    private static boolean conflictingStartOrEndDate(Scheduling scheduling, Scheduling schedulingToCheck) {
```

Comment:
I don't know if the name os this method is right, maybe: isThereAnyDateConflicts? (I don't know which one is worst)

Reply:
I think this one is ok